File: Backend/htbp_parent/common/common-utils/src/main/java/com/seclib/htbp/common/exception/GlobalExceptionHandler.java
package com.seclib.htbp.common.exception;

import com.seclib.htbp.common.result.Result;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseBody;

@ControllerAdvice
public class GlobalExceptionHandler {
    @ExceptionHandler(Exception.class)
    @ResponseBody
    public Result error(Exception e){
        e.printStackTrace();
        return Result.fail();
    }

    @ExceptionHandler(HtbpException.class)
    @ResponseBody
    public Result error(HtbpException e){
        e.printStackTrace();
        return Result.fail();
    }

}

File: Backend/htbp_parent/common/common-utils/src/main/java/com/seclib/htbp/common/exception/GlobalExceptionHandler.kt
package com.seclib.htbp.common.exception

import org.springframework.web.bind.annotation.ControllerAdvice
import java.lang.Exception
import org.springframework.web.bind.annotation.ResponseBody
import com.seclib.htbp.common.exception.HtbpException
import com.seclib.htbp.common.result.Result
import io.swagger.annotations.ApiModel
import java.lang.RuntimeException
import io.swagger.annotations.ApiModelProperty
import org.springframework.web.bind.annotation.ExceptionHandler

@ControllerAdvice
open class GlobalExceptionHandler {
    @ExceptionHandler(Exception::class)
    @ResponseBody
    open fun error(e: Exception): Result<*> {
        e.printStackTrace()
        return Result.fail<Any>()
    }

    @ExceptionHandler(HtbpException::class)
    @ResponseBody
    open fun error(e: HtbpException): Result<*> {
        e.printStackTrace()
        return Result.fail<Any>()
    }
}